src/compiler/factory/node_factory.rs

```diff
@@ -73,9 +73,9 @@ impl NodeFactory {
         &self,
         base_factory: &TBaseNodeFactory,
         token: SyntaxKind,
-    ) -> Token {
+    ) -> BaseNode {
```

Comment:
I ran into an issue with my `Token`/`Expression` as enum variants of `Node` hierarchy - Typescript blurs the line a bit eg calling `createTokenNode<PrimaryExpression>()`. So I preserved `Expression` but removed `Token` in favor of just `BaseNode` as a variant of `Node` (and to represent "basic" concrete expression types like boolean literals, I added a `TokenExpression` variant to `Expression`). So that way these things that were returning `Token` can return `BaseNode`, which can then be hoisted into an `Expression::TokenExpression()` (when we're constructing expressions like boolean literals)